### PR TITLE
Sample playbook and module to install ftd

### PR DESCRIPTION
**Pull request by surbhibansal14**
Created a sample playbook test_kick.yml and module library/generated/install_ftd.py to test integration of Ansible with KICK.

### CODE REVIEW

File: library/ftd_install.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+
+def check_that_update_is_needed(module, system_info):
+    target_ftd_version = module.params["image_version"]
+    if target_ftd_version == system_info['softwareVersion']:
+        module.exit_json(changed=False, msg="FTD already has %s version of software installed." % target_ftd_version)
+
+
+def check_management_and_dns_params(resource, params):
+    if not params.get('device_ip') or not params.get('device_netmask') or not params.get('device_gateway'):
```
Comment (119Vik) [inline library/ftd_install.py]: will be simplier to use
`if not all([a.get('a'), b.get('b')])`

File: library/ftd_install.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+        management_ip = resource.execute_operation(GET_MANAGEMENT_IP_LIST_OPERATION, {})['items'][0]
+        params['device_ip'] = params.get('device_ip') or management_ip['ipv4Address']
```
Comment (119Vik) [inline library/ftd_install.py]: can't we just pass default argument to **.get** method here?
`        params['device_ip'] = params.get('device_ip', management_ip['ipv4Address'])
`

File: samples/install_ftd.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+        console_ip: 10.89.0.0
+        console_port: 2004
+        console_username: console_user
+        console_password: console_pass
+
+        tftp_server: 10.89.0.11
+        rommon_file_location: 'installers/ftd-boot-9.10.1.3.lfbff'
+        image_file_location: 'https://10.89.0.11/installers/ftd-6.3.0-83.pkg'
+        image_version: 6.3.0-83
```
Comment (119Vik) [inline samples/install_ftd.yml]: let's add last empty line here

File: library/ftd_install.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    HAS_KICK = True
+except ImportError:
+    HAS_KICK = False
+
+try:
+    from ansible.module_utils.configuration import BaseConfigurationResource, ParamName, PATH_PARAMS_FOR_DEFAULT_OBJ
+except ImportError:
+    from module_utils.configuration import BaseConfigurationResource, ParamName, PATH_PARAMS_FOR_DEFAULT_OBJ
+
+GET_SYSTEM_INFO_OPERATION = 'getSystemInformation'
```
Comment (119Vik) [inline library/ftd_install.py]: Can we create Enum like class FTDOperations which will wrap all Operations?

Comment (annikulin) [reply library/ftd_install.py]: Good point, updated.

File: library/ftd_install.py (hunk 2 above)
Comment (119Vik) [reply library/ftd_install.py]: Ok. got it

**Comment by 119Vik** [comment]
looks good for me.

**Pull request merged**